### DISCUSSION TITLE
Musings about reproducing history

### DISCUSSION THREAD

**Discussion by joehays** [lava-nc/lava#138] [General]
I'd just like to share a thought for the Lava/Magma team to consider. Given a limited knowledge of Lava and the team's plans... these musings may be off the mark.

One objective of ROS2 was to stop managing the communication middleware (or software bus), so ROS2 embraced DDS (a proven standard for "real-time" inter-process and networked node communication). Here is the spec,

https://www.omg.org/spec/DDS/1.4

It *feels* like the Magma CSP is going down the same road that ROS1 did. Once Lava leaves the research/laboratory/early adopters, it runs the risk of being difficult to integrate into industrial applications if the CSP is not compliant with other standards. So, my recommendation is that the Lava/Magma team seriously consider embracing a standard, like DDS, for the backend implementation of CSP.

**Comment by awintel** [comment]
Thanks Joe. We will look into this.
